monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,379 @@
+
+  @mock.patch('os.access')
+  def testunit_assert_collectd_responds_WHEN_filesystem_is_readonly(
+      self, mock_access):
+    # Mock out os.access to guarantee that write access is rejected.
+    mock_access.return_value = False
+
+    self.assertRaises(
+        check_collectd_mlab.ReadonlyFilesystemCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+    self.assertTrue(mock_access.called)
+
+  def testunit_assert_collectd_responds_WHEN_sock_connect_fails(self):
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.SocketConnectionCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_backend_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysBackendCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_slice_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysFrontendCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_acl_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = os.path.join(
+        self._testdata_dir, 'fake_slice')
+    check_collectd_mlab.VSYSPATH_ACL = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysAclCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_acl_incomplete(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = os.path.join(
+        self._testdata_dir, 'fake_slice')
+    check_collectd_mlab.VSYSPATH_ACL = os.path.join(
+        self._testdata_dir, 'fake_acl')
+
+    self.assertRaises(
+        check_collectd_mlab.MissingSliceFromVsysAclCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  @mock.patch('subprocess.Popen')
+  def testcover_run_collectd_nagios(self, mock_popen):
+    mock_popen.return_value.wait.return_value = 2
+
+    returned_value = check_collectd_mlab.run_collectd_nagios(
+        'host', 'metric', 'value', 'warning', 'critical')
+
+    self.assertEqual(returned_value, 2)
+    self.assertTrue(mock_popen.called)
+
+  @mock.patch('check_collectd_mlab.run_collectd_nagios')
+  def testcover_assert_collectd_nagios_levels(self, mock_run_collectd_nagios):
```

Comment:
I made up the name. This is probably the lowliest of tests. It makes sure that the interpreter visits every line even when no functionality in particular is being tested. The best (not necessarily good) defense for this approach is that since python doesn't do static type checking, "silly" errors may not show up until run time (a variable is used before being defined, or a typo makes a variable appear to not exist). These tests give "silly" mistakes a chance to show up beforehand.